manifest-build-tools/lib/RepositoryOperator.py

```diff
@@ -317,6 +317,16 @@ def get_lastest_commit_id(self, repo_dir):
         else:
             raise RuntimeError("Unable to get commit id in directory {0}".format(repo_dir))
 
+    def get_lastest_merge_commit_before_date(self, repo_dir, date):
+        if repo_dir is None or not os.path.isdir(repo_dir):
+            raise RuntimeError("The repository directory is not a directory")
+        return_code, output, error = self.git.run(['log', '--merges', '--format=format:%H', '--before='+date, '-n', '1'], directory=repo_dir)
```

Comment:
Thanks. I was worrying the ```git log --before="datestring"```,  there should be quote around $datestring.
but Tian told me the self.git.run() will add quote(") around those element . so it's fine.